src/historywork/DownloadBucketsWork.cpp

```diff
@@ -17,8 +17,8 @@ namespace stellar
 DownloadBucketsWork::DownloadBucketsWork(
     Application& app, std::map<std::string, std::shared_ptr<Bucket>>& buckets,
     std::vector<std::string> hashes, TmpDir const& downloadDir,
-    std::shared_ptr<HistoryArchive> archive)
-    : BatchWork{app, "download-verify-buckets"}
+    std::shared_ptr<HistoryArchive> archive, std::function<bool()> verifyCb)
```

Comment:
I think it's time to actually do https://github.com/stellar/stellar-core/pull/3030#discussion_r628649644 what do you think?

Reply:
I added a new small work class `WorkWithCallback` to address this, let me know what you think.